### PR TITLE
fix(IV): react IV demo adjustments

### PR DESCRIPTION
**Pull request by jonh-1**
* align project to work with pnpm (this seems to fix the build issue)
* refactor nested ternary

### CODE REVIEW

**Review by jameson-w-taylor — APPROVED** [review]

File: identity-vault/getting-started/src/util/session-vault.ts [hunk 1]
```diff
@@ -71,25 +68,28 @@ export const clearSession = async (): Promise<void> => {
...
-      ? DeviceSecurityType.Both
-      : DeviceSecurityType.None;
-  await vault.updateConfig({
-    ...(vault.config as IdentityVaultConfig),
-    type,
-    deviceSecurityType,
-  });
+  const newConfig = { ...(vault.config as IdentityVaultConfig) };
+
+  switch (mode) {
```
Comment (jameson-w-taylor) [inline identity-vault/getting-started/src/util/session-vault.ts:73]: This is missing the `deviceSecurityType` setting, from the previous version of this code.

See this snippet:
```ts
export const updateUnlockMode = async (mode: UnlockMode): Promise<void> => {
  const newConfig = { ...(valut.config as IdentityVaultConfig) };

  switch (mode) {
    case 'BiometricsWithPasscode': {
      newConfig.type = VaultType.DeviceSecurity;
      newConfig.deviceSecurityType = DeviceSecurityType.Both;
      break;
    }
    case 'InMemory': {
      newConfig.type = VaultType.InMemory;
      newConfig.deviceSecurityType = DeviceSecurityType.None;
      break;
    }
    default: {
      newConfig.type = VaultType.SecureStorage;
      newConfig.deviceSecurityType = DeviceSecurityType.None;
      break;
    }
  }

  await vault.updateConfig(newConfig);
};
```

**Review by jameson-w-taylor — CHANGES_REQUESTED** [review]

File: identity-vault/getting-started/src/util/session-vault.ts (hunk 1 above)
Comment (jonh-1) [reply identity-vault/getting-started/src/util/session-vault.ts:73]: Shoot, thanks for catching

**Review by jameson-w-taylor — APPROVED** [review]

**Review by kensodemann — APPROVED** [review]
LGTM

File: identity-vault/getting-started/android/app/capacitor.build.gradle [hunk 2]
```diff
@@ -15,7 +15,7 @@ dependencies {
     implementation project(':capacitor-status-bar')
 
 }
-apply from: "../../node_modules/@ionic-enterprise/identity-vault/src/android/ionicnativeauth.gradle"
+apply from: "../../../../node_modules/.pnpm/@ionic-enterprise+identity-vault@5.12.4_typescript@5.3.3/node_modules/@ionic-enterprise/identity-vault/src/android/ionicnativeauth.gradle"
```
Comment (eric-horodyski) [inline identity-vault/getting-started/android/app/capacitor.build.gradle:18]: This file path looks odd to me. @jameson-w-taylor @kensodemann thoughts?

File: identity-vault/getting-started/android/capacitor.settings.gradle [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-project(':capacitor-haptics').projectDir = new File('../node_modules/@capacitor/haptics/android')
+project(':capacitor-haptics').projectDir = new File('../../../node_modules/.pnpm/@capacitor+haptics@5.0.7_@capacitor+core@5.7.0/node_modules/@capacitor/haptics/android')
 
 include ':capacitor-keyboard'
-project(':capacitor-keyboard').projectDir = new File('../node_modules/@capacitor/keyboard/android')
+project(':capacitor-keyboard').projectDir = new File('../../../node_modules/.pnpm/@capacitor+keyboard@5.0.8_@capacitor+core@5.7.0/node_modules/@capacitor/keyboard/android')
 
 include ':capacitor-status-bar'
-project(':capacitor-status-bar').projectDir = new File('../node_modules/@capacitor/status-bar/android')
+project(':capacitor-status-bar').projectDir = new File('../../../node_modules/.pnpm/@capacitor+status-bar@5.0.7_@capacitor+core@5.7.0/node_modules/@capacitor/status-bar/android')
```
Comment (eric-horodyski) [inline identity-vault/getting-started/android/capacitor.settings.gradle:15]: Same here.

File: identity-vault/getting-started/ios/App/Podfile
Comment (eric-horodyski) [inline identity-vault/getting-started/ios/App/Podfile:1]: And here as well

File: identity-vault/getting-started/android/app/capacitor.build.gradle (hunk 2 above)
Comment (eric-horodyski) [reply identity-vault/getting-started/android/app/capacitor.build.gradle:18]: Nevermind, look's like it's a PNPM thing. Interesting. Looks weird to me!

**Review by eric-horodyski — APPROVED** [review]

**Pull request merged**